File: backend/app/schemas/paper_generation.py
from pydantic import BaseModel, Field


class PaperGenerationRequest(BaseModel):
    title: str
    experiment_ids: list[int] = Field(default_factory=list)
    selected_node_ids: list[int] = Field(default_factory=list)
    target_venue: str = "CVPR"
    main_metric_names: list[str] = Field(default_factory=list)
    primary_dataset_names: list[str] = Field(default_factory=list)
    prompt: str | None = None
    model_name: str | None = None
    use_llm: bool = True
    try_compile_pdf: bool = True


class PaperGenerationResponse(BaseModel):
    title: str
    target_venue: str
    output_dir: str
    tex_path: str
    figure_paths: list[str]
    table_paths: list[str]
    used_llm: bool
    pdf_compiled: bool
    pdf_path: str | None = None
    pdf_url: str | None = None
    output_url_base: str | None = None
    tex_url: str | None = None
    context_snapshot_url: str | None = None
    compile_log: str | None = None
    sections: dict[str, str]

File: frontend/src/services/paperGeneration.ts
import { api } from './api'

export interface GeneratePaperPayload {
  title: string
  experiment_ids: number[]
  selected_node_ids: number[]
  target_venue?: string
  main_metric_names?: string[]
  prompt?: string
  model_name?: string
  use_llm?: boolean
  try_compile_pdf?: boolean
}

export interface GeneratedPaperDraft {
  title: string
  target_venue: string
  output_dir: string
  tex_path: string
  figure_paths: string[]
  table_paths: string[]
  used_llm: boolean
  pdf_compiled: boolean
  pdf_path?: string | null
  pdf_url?: string | null
  output_url_base?: string | null
  tex_url?: string | null
  context_snapshot_url?: string | null
  sections: Record<string, string>
}

export async function generatePaper(payload: GeneratePaperPayload) {
  const { data } = await api.post<GeneratedPaperDraft>('/paper-generation/generate', payload)
  return data
}
